packages/core/src/api/guild.ts

```diff
@@ -0,0 +1,869 @@
+		return this.rest.patch(Routes.guildWidgetSettings(guildId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPatchAPIGuildWidgetSettingsResult>;
+	}
+
+	/**
+	 * Fetches the widget for a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the widget for
+	 */
+	public async getWidget(guildId: Snowflake) {
+		return this.rest.get(Routes.guildWidgetJSON(guildId)) as Promise<RESTGetAPIGuildWidgetJSONResult>;
+	}
+
+	/**
+	 * Fetches the vanity url for a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the vanity url for
+	 */
+	public async getVanityURL(guildId: Snowflake) {
+		return this.rest.get(Routes.guildVanityUrl(guildId)) as Promise<RESTGetAPIGuildVanityUrlResult>;
+	}
+
+	/**
+	 * Fetches the widget image for a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the widget image for
+	 * @param style - The style of the widget image
+	 */
+	public async getWidgetImage(guildId: Snowflake, style?: GuildWidgetStyle) {
+		return this.rest.get(Routes.guildWidgetImage(guildId), {
+			query: makeURLSearchParams({ style }),
+		}) as Promise<RESTGetAPIGuildWidgetImageResult>;
+	}
+
+	/**
+	 * Fetches the welcome screen for a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the welcome screen for
+	 */
+	public async getWelcomeScreen(guildId: Snowflake) {
+		return this.rest.get(Routes.guildWelcomeScreen(guildId)) as Promise<RESTGetAPIGuildWelcomeScreenResult>;
+	}
+
+	/**
+	 * Edits the welcome screen for a guild
+	 *
+	 * @param guildId - The id of the guild to edit the welcome screen for
+	 * @param data - The new welcome screen data
+	 * @param reason - The reason for editing the welcome screen
+	 */
+	public async editWelcomeScreen(guildId: Snowflake, data?: RESTPatchAPIGuildWelcomeScreenJSONBody, reason?: string) {
+		return this.rest.patch(Routes.guildWelcomeScreen(guildId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPatchAPIGuildWelcomeScreenResult>;
+	}
+
+	/**
+	 * Edits a user's voice state in a guild
+	 *
+	 * @param guildId - The id of the guild to edit the current user's voice state in
+	 * @param userId - The id of the user to edit the voice state for
+	 * @param data - The data for editing the voice state
+	 * @param reason - The reason for editing the voice state
+	 */
+	public async editUserVoiceState(
+		guildId: Snowflake,
+		userId: Snowflake,
+		data: RESTPatchAPIGuildVoiceStateUserJSONBody,
+		reason?: string,
+	) {
+		await this.rest.patch(Routes.guildVoiceState(guildId, userId), { reason, body: data });
+	}
+
+	/**
+	 * Fetches all emojis for a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the emojis for
+	 */
+	public async getEmojis(guildId: Snowflake) {
+		return this.rest.get(Routes.guildEmojis(guildId)) as Promise<RESTGetAPIGuildEmojisResult>;
+	}
+
+	/**
+	 * Fetches an emoji for a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the emoji for
```

Comment:
Wouldn't it be "from", since it indicates where the emoji is from?
```suggestion
	 * @param guildId - The id of the guild to fetch the emoji from
```

If so, this request probably applies to other places.